### PR TITLE
Display latencies for getting a connection from Hikari pool.

### PR DESCRIPTION
**Pull request by psudheer21**
The new output:
01:39:14,402 (DBWorkload.java:1016) INFO  - Num New Order transactions : 36, time seconds: 180
01:39:14,402 (DBWorkload.java:1017) INFO  - TPM-C: 12
01:39:14,402 (DBWorkload.java:1018) INFO  - Efficiency : 93.31%
01:39:14,403 (DBWorkload.java:1056) INFO  - NewOrder, Avg Latency: 35.67194444444444 msecs, p99 Latency: 172.218 msecs Whole operation, Avg Latency: 56.37466666666666 msecs, p99 Latency: 231.338 msecs
01:39:14,404 (DBWorkload.java:1056) INFO  - Payment, Avg Latency: 25.760722222222224 msecs, p99 Latency: 216.9 msecs Whole operation, Avg Latency: 39.668416666666666 msecs, p99 Latency: 255.616 msecs
01:39:14,404 (DBWorkload.java:1056) INFO  - OrderStatus, Avg Latency: 19.56925 msecs, p99 Latency: 36.103 msecs Whole operation, Avg Latency: 20.107 msecs, p99 Latency: 36.516 msecs
01:39:14,404 (DBWorkload.java:1056) INFO  - Delivery, Avg Latency: 63.89 msecs, p99 Latency: 87.842 msecs Whole operation, Avg Latency: 64.649 msecs, p99 Latency: 88.568 msecs
01:39:14,404 (DBWorkload.java:1056) INFO  - StockLevel, Avg Latency: 135.317 msecs, p99 Latency: 136.352 msecs Whole operation, Avg Latency: 135.9805 msecs, p99 Latency: 136.877 msecs
01:39:14,404 (DBWorkload.java:1068) INFO  - Acquire Connection, Avg Latency: 19.240474999999996 msecs, p99 Latency: 155.616 msecs

Reviewers:
Mihnea, Rob

### CODE REVIEW

File: src/com/oltpbenchmark/api/Worker.java [hunk 1]
```diff
@@ -420,7 +439,10 @@ protected final TransactionType doWork(boolean measure, SubmittedProcedure piece
                     // }
 
                     status = TransactionStatus.UNKNOWN;
+
+                    startOperation.setValue(System.nanoTime());
                     status = this.executeWork(conn, next);
```
Comment (robertsami) [inline src/com/oltpbenchmark/api/Worker.java:524]: what if we get a retry status?

File: src/com/oltpbenchmark/api/Worker.java [hunk 2]
```diff
@@ -310,10 +318,14 @@ public final void run() {
                 }
             }
 
-            long startOperation = System.nanoTime();
+            MutableLong startOperation = new MutableLong();
+            MutableLong endOperation = new MutableLong();
+            MutableLong startConnection = new MutableLong();
+            MutableLong endConnection = new MutableLong();
```
Comment (robertsami) [inline src/com/oltpbenchmark/api/Worker.java]: could we just pass in startConnection and endConnection, keep the old startOperation/endOperation computation outside here, and compute the difference if desired?

Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: I would prefer this, cuz the intention is clearer here. where we try to get the time of the actual operation execution.
Let me know what you think

File: src/com/oltpbenchmark/api/Worker.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java:524]: It was a conscious decision to just include the time of the actual operation and not the failure cases.
Let me know what you think.

Comment (robertsami) [reply src/com/oltpbenchmark/api/Worker.java:524]: i do think its useful to isolate measurement to successful operations in terms of our own understanding, as you did here, but we should also be measuring the latency a user would experience for said operation, from op start, through each retry, to final completion, when grading ourselves

File: src/com/oltpbenchmark/api/Worker.java (hunk 2 above)
Comment (robertsami) [reply src/com/oltpbenchmark/api/Worker.java]: yeah i dont disagree. i might consider returning something richer than TransactionType that includes latency info as well, to further clarify/document the intention

File: src/com/oltpbenchmark/DBWorkload.java [hunk 3]
```diff
@@ -1018,32 +1018,52 @@ private static Results runWorkload(List<BenchmarkModule> benchList, boolean verb
...
     for (Worker<?> w : workers) {
       for (LatencyRecord.Sample sample : w.getLatencyRecords()) {
         list_latencies.get(sample.tranType - 1).add(sample.operationLatencyUs);
       }
+
+      // Since we have 5 transaction types, use the index 5 to store the latency for the Hikari pool
+      // connections.
+      if (printConnectionLatency) {
+        for (LatencyRecord.Sample sample : w.getConnectionLatencyRecords()) {
+          list_latencies.get(5).add(sample.operationLatencyUs);
```
Comment (robertsami) [inline src/com/oltpbenchmark/DBWorkload.java]: this feels hacky. is there some alternative approach where we can track `overall_latencies` as the original latency semantic (including connection acquisition and all retries), `acq_connection_latencies`as the time taken to acquire a connection, and `final_op_try_latency` as the latency of the final/successful op only (as in excluding retries), and each list can be of size `final int kNumTxnTypes = 5;`?

Comment (robertsami) [reply src/com/oltpbenchmark/DBWorkload.java]: so, going further, each worker can have getLatencyRecords , getConnectionLatencies, and getFinalOpLatencies

File: src/com/oltpbenchmark/api/Worker.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java:524]: But 1% of the transactions have failures by design.
Should we include those?

File: src/com/oltpbenchmark/DBWorkload.java (hunk 3 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/DBWorkload.java]: Changed the code to be easier to read by making the intentions clearer.

File: src/com/oltpbenchmark/api/Worker.java (hunk 2 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: Thats a good idea. Let me incorporate that.

File: src/com/oltpbenchmark/api/Worker.java (hunk 1 above)
Comment (robertsami) [reply src/com/oltpbenchmark/api/Worker.java:524]: id say lets match the previous behavior for those?

File: src/com/oltpbenchmark/DBWorkload.java [hunk 4]
```diff
@@ -1113,7 +1135,9 @@ public static void mergeResults(String dirPath, String[] fileNames) {
     LOG.info("Num New Order transactions : " + numNewOrderTransactions + ", time seconds: " + time);
     LOG.info("TPM-C: " + df.format(tpmc));
     LOG.info("Efficiency : " + df.format(efficiency) + "%");
-    printLatencies(list_latencies, transactionTypes);
+    for (int i = 0; i < 5; ++i) {
```
Comment (robertsami) [inline src/com/oltpbenchmark/DBWorkload.java]: `i < list_latencies.size()`?

File: src/com/oltpbenchmark/api/Worker.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java:524]: Done.
I have changed the behaviour a bit and showed the actual latency by default and the whole operation latency along with the connection acquisition latency protected by the flag.

File: src/com/oltpbenchmark/DBWorkload.java (hunk 4 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/DBWorkload.java]: Done.

File: src/com/oltpbenchmark/api/Worker.java [hunk 5]
```diff
@@ -352,15 +423,35 @@ public final void run() {
             long end = System.nanoTime();
             postState = wrkldState.getGlobalState();
 
+
```
Comment (robertsami) [inline src/com/oltpbenchmark/api/Worker.java]:
```suggestion
```

**Review by robertsami — APPROVED** [review]

File: src/com/oltpbenchmark/api/Worker.java (hunk 5 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/api/Worker.java]: Done.

**Pull request merged**